### PR TITLE
IDEX-4097: Document che.properties

### PR DESCRIPTION
**Pull request by TylerJewell**
@skabashnyuk 
@vparfonov 
@garagatyi 

wdyt?

### CODE REVIEW

**Comment by benoitf** [comment]
Need to remove https://github.com/codenvy/che/blob/master/assembly-main/src/assembly/conf/che.properties as well

**Comment by TylerJewell** [comment]
Yes, good catch.  This is done.

File: assembly-ide-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties [hunk 1]
```diff
@@ -9,84 +9,128 @@
...
-# path to templates' description folder
-project.template_descriptions_dir=${che.home}/templates
-# path to templates' location
-project.template_location_dir=${che.home}/templates
+# URL path to api service.  
+# Browser clients use this to initiate REST communiations with workspace master
+api.endpoint=http://localhost:${SERVER_PORT}/ide/api
 
-user.self.creation.allowed=true
+#### Everrest is a Java Web Services toolkit that manages JAX & web socket communications
```
Comment (benoitf) [inline assembly-ide-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties]: I would call it JAX-RS, not JAX

**Comment by vparfonov** [comment]
ok

**Comment by TylerJewell** [comment]
Updates made.  @skabashnyuk?

File: assembly-ide-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties [hunk 2]
```diff
@@ -9,84 +9,128 @@
...
+# Che injects this archive into machines when they are booted or started.
 machine.server.ext.archive=${che.home}/lib/ws-agent.zip
 
-# machine terminal server archive
+# The location of the Web Socket terminal used within the browser.
+# This is copied into the machine and run from within it.
 machine.server.terminal.archive=${che.home}/lib/terminal
+
+# The machine's log files are stored here
+machine.logs.location=${catalina.base}/logs
```
Comment (skabashnyuk) [inline assembly-ide-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties]: machine.logs.location=${che.logs.dir}/machine/logs ?

File: assembly-ide-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties [hunk 3]
```diff
@@ -9,84 +9,128 @@
...
-#token uri
 oauth.github.tokenuri= https://github.com/login/oauth/access_token
-#redirected uris
 oauth.github.redirecturis= http://localhost:${SERVER_PORT}/ide/api/oauth/callback
+auth.oauth.access_denied_error_page=/error-oauth
+
+# Activates user self-service at the API level.
+# Che has a single identity implementation, so this does not change the user experience.
+# If true, enables user creation at API level
+user.self.creation.allowed=true
```
Comment (skabashnyuk) [inline assembly-ide-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties]: user.self.creation.allowed=false?

File: assembly-ide-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties (hunk 2 above)
Comment (TylerJewell) [inline assembly-ide-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties]: +1

File: assembly-ide-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties (hunk 3 above)
Comment (TylerJewell) [inline assembly-ide-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties]: +1

**Pull request merged**